File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
import java.util.ArrayList;
import java.util.List;
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double sum=0;
        double maxsum=Integer.MIN_VALUE;
        List<Integer> arr=new ArrayList<>();
        for(int i=0;i<nums.length;i++){
           arr.add(nums[i]);
           sum+=nums[i];
           if(arr.size()==k){
                maxsum=Math.max(maxsum,sum);
                sum=sum-arr.remove(0);
           }
        }
        return maxsum/k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        
        # maxi = 0
        sumx = 0
        for i in range(k):
            sumx+=nums[i]
        maxi = sumx/k

        for i in range(k,len(nums)):
            sumx-=nums[i-k]
            sumx+=nums[i]
            maxi = max(maxi,sumx/k)
        return maxi